include/zephyr/drivers/spi.h

```diff
@@ -229,39 +229,27 @@ struct spi_cs_control {
  *             spidev: spi-device@0 { ... };
  *     };
  *
- * Assume that @p gpio0 follows the standard convention for specifying
- * GPIOs, i.e. it has the following in its binding:
- *
- *     gpio-cells:
- *     - pin
- *     - flags
- *
  * Example usage:
  *
- *     struct spi_cs_control *ctrl =
- *             SPI_CS_CONTROL_PTR_DT(DT_NODELABEL(spidev), 2);
+ *     struct spi_cs_control ctrl =
+ *             SPI_CS_CONTROL_INIT(DT_NODELABEL(spidev), 2);
  *
  * This example is equivalent to:
  *
- *     struct spi_cs_control *ctrl =
- *             &(struct spi_cs_control) {
- *                     .gpio_dev = DEVICE_DT_GET(DT_NODELABEL(gpio0)),
- *                     .delay = 2,
- *                     .gpio_pin = 1,
- *                     .gpio_dt_flags = GPIO_ACTIVE_LOW
- *             };
- *
- * This macro is not available in C++.
+ *     struct spi_cs_control ctrl = {
+ *             .gpio = SPI_CS_GPIOS_DT_SPEC_GET(DT_NODELABEL(gpio0)),
+ *             .delay = 2,
+ *     };
  *
  * @param node_id Devicetree node identifier for a device on a SPI bus
  * @param delay_ The @p delay field to set in the @p spi_cs_control
  * @return a pointer to the @p spi_cs_control structure
  */
-#define SPI_CS_CONTROL_PTR_DT(node_id, delay_)			  \
-	(&(struct spi_cs_control) {				  \
```

Comment:
Just caught this with the arch meeting and wanted to mention it here too.

The old spi macro declaration appears to be a statement expr(ession). It's a gcc-ism (just happens to be supported by clang too), but they aren't ISO C or C++. We (somewhat unfortunately) do have them all over the place still.

I would say that the preferred way is to embed the struct, which is what Gerrard has done. We pay the price in terms of an extra 4 bytes, but it's at least ISO C / C++.